### PR TITLE
Add File Upload Push Service

### PR DESCRIPTION
**Pull request by luvletter2333**
Sql Update Scriprt:

alter table [user] add FilePush int null default 0
update [User] set FilePush=0

### CODE REVIEW

File: App_code/SAAO/User.cs [hunk 1]
```diff
@@ -166,6 +183,7 @@ public User(string username)
             _mail = dr["mail"].ToString();
             _phone = dr["phone"].ToString();
             _wechat = dr["wechat"].ToString();
+            _filepush = (int)dr["FilePush"];
```
Comment (tonnylyz) [inline App_code/SAAO/User.cs:188]: reader.GetInt32(reader.GetOrdinal(columnName))
or

    Convert.ToInt32(reader[columnName])

is recommanded.

**Review by tonnylyz — COMMENTED** [review]
Some small problems:
 * Has it been tested in server side? (pull it in server /test/ and test)
 * You can commit the database change on server since it doesn't conflit with current version
 * Use Automatic code formatting in Visual Studio before commit
 * Code Identifier naming respect a set of specific rules for C# (e.g. camel notation)
 * Adopt some C#6 language features (e.g. string interpolation)

**Comment by luvletter2333** [comment]
server side test pass
umm i almost don't learn c#6 features

File: App_code/Handler/FileHandler.cs [hunk 2]
```diff
@@ -52,37 +50,29 @@ public override void Process(System.Web.HttpContext context)
...
                         {
-                            ["touser"] = rec,
-                            ["msgtype"] = "file",
-                            ["agentid"] = 4,
-                            ["file"] = new JObject { ["media_id"] = file.MediaId }
-                        };
-                        SAAO.Utility.SendMessgaeBySAAOHelper(oFile);
-                        SAAO.Utility.SendMessgaeBySAAOHelper(oText);
-                    }
+                            ["content"] = string.Format("新文件提醒：\n文件名：{0}\n上传者：{1}\n上传时间：{2}\n备注：{3}",file.Name,file.Uploader.Realname, string.Format("{0:f}", file.UploadTime),file.Info)
```
Comment (tonnylyz) [inline App_code/Handler/FileHandler.cs]: Using string string interpolation(C#6):
```
["content"] = $"新文件提醒：\n文件名：{file.Name}\n上传者：{file.Uploader.Realname}\n上传时间：{file.UploadTime.ToString("f")}\n备注：{file.Info}"
```

File: App_code/SAAO/File.cs [hunk 3]
```diff
@@ -323,7 +323,7 @@ public static JArray ListJson(DateTime start, DateTime end)
             si.AddParameter("@start", SqlIntegrate.DataType.Date, start);
             si.AddParameter("@end", SqlIntegrate.DataType.Date, end);
             var dt = si.Adapter("SELECT [File].*, [User].[realname], [User].[group] FROM [File] INNER JOIN [User] ON [File].[uploader] = [User].[UUID] AND [File].[uploadTime] BETWEEN @start AND @end ORDER BY [File].[ID] DESC");
-            var a = new JArray();
+             var a = new JArray();
```
Comment (tonnylyz) [inline App_code/SAAO/File.cs:326]: Mysterious space?

**Review by tonnylyz — APPROVED** [review]
Nice job!
Btw 我觉得用dotnet core取代现有的framework的可行性是具备的。如果有兴趣继续改造SAAO的话，可能需要在实用性功能（比如和微信的深度整合，毕竟希望能发挥更大的用途而不是满足自己的技术热情）以及一些技术简化（方便后续的低门槛开发）做一些改进工作。
现有的仿/类MVC的架构某些地方过于繁复和啰嗦。
EntityFramework在纯微软的架构上也是十分有优势的，这个可以试试看。

File: App_code/SAAO/File.cs (hunk 3 above)
Comment (luvletter2333) [reply App_code/SAAO/File.cs:326]: ummm
调试那个时间bug时删代码漏了点东西

**Comment by luvletter2333** [comment]
emm其实我想过在企业微信直接上传文件
也是主要针对ios用户考虑
理论上设置好回调就可以了
但是iOS上面没法直接向saao助手发文件的

**Pull request merged**